### PR TITLE
CB-9598 Switching to use `cordova-common` module

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
See [CB-9598](https://issues.apache.org/jira/browse/CB-9598) for details.

This PR just removes code which has been copied previously to `cordova-common` and updates requires to point to that code. The LIB  includes `cordova-common`  as a bundled dependency, so it doesn't need to be published.

To npm links to new module could be set up using `coho npm-link` after pulling in the following PR to cordova-coho: https://github.com/apache/cordova-coho/pull/94

### CODE REVIEW

**Comment by nikhilkh** [comment]
Looks like there is a Travis CI failure and appveyor does not even run the tests.

**Comment by vladimir-kotikov** [comment]
@nikhilkh, please review once more. The failing tests was fixed. 
ConfigChanges also was reworked in 2e11c52 as per [conversation on dev thread](http://callback.markmail.org/thread/pmtqf6e5yxvmvt7w)

File: cordova-lib/package.json [hunk 1]
```diff
@@ -40,14 +38,19 @@
...
+  "bundledDependencies": [
+    "cordova-common"
+  ],
   "devDependencies": {
     "istanbul": "^0.3.4",
     "jasmine-node": "1.14.5",
     "jshint": "2.5.8",
-    "rewire": "2.1.3"
+    "rewire": "2.1.3",
+    "osenv": "0.1.0",
```
Comment (nikhilkh) [inline cordova-lib/package.json]: why are thee dev dependecies now?

**Comment by nikhilkh** [comment]
LGTM overall.

File: cordova-lib/package.json (hunk 1 above)
Comment (vladimir-kotikov) [inline cordova-lib/package.json]: My bad. Returned plist and xcode back to 'dependencies' section and totally got rid of osenv module.

**Comment by omefire** [comment]
:nit Looks like we need to update 'cordova-lib' to 'cordova-common' in the README.md file : https://github.com/apache/cordova-lib/commit/28ce0d1d8665caff4977622f6b178b7f4899896e#diff-fe6330ba7c70b61a5127352c5cdb9e32R22

**Pull request merged**